Enable modifying feature flags based on URL query parameters

nylen opened the pull request:
This will enable us to modify feature flags without restarting the Webpack build and without modifying any files locally.

Since calypso.live [runs Calypso in development mode](https://github.com/Automattic/calypso-live-branches/blob/09de597/calypso.json#L7) we can use it there too.

When loading a fresh Calypso page, you can specify a `?flags=` query argument.  Here are a couple of examples:

- `?flags=flag1`: Enable feature `flag1`.
- `?flags=-flag2`: Disable feature `flag2`.
- `?flags=+flag1,-flag2`: Enable feature `flag1` and disable feature `flag2`.

samouri commented:
Thank you for the PR! This could definitely be helpful in a ton of circumstances (like disabling forced-sympathy for one).

The bad part is that it won't work on any server-side features. 🤷‍♂️

nylen commented:
> Would it be better to use `console.warn` to get the dev's attention a bit more?

I tried this, but it looks a bit weird with the stack trace available and with the other things using `console.warn` being actually warnings (HMR needing refresh, etc.)  I changed it to just use a bold font for those messages.

> Should we consider allowing this for staging?

Done.

> The bad part is that it won't work on any server-side features.

Yep, for that you'll need to use `ENABLE_FEATURES` and/or `DISABLE_FEATURES`.  I added a note to the readme accordingly.
